client/src/featureform/train_test_split.py

```diff
@@ -0,0 +1,212 @@
+        self.request_queue.append(req)
+        try:
+            response = next(self.response_stream)
+            return response
+        except grpc.RpcError as e:
+            print(f"Caught RPC error of type: {e.code()} - {e.details()}")
+            if e.code() == grpc.StatusCode.RESOURCE_EXHAUSTED:
+                print("Resource exhausted")
+            raise
+
+    def _create_request_generator(self):
+        """Yields requests from the queue for the gRPC stream."""
+        while True:
+            if self.request_queue:
+                yield self.request_queue.popleft()
+
+
+@dataclass
+class _NpProtoTypeParser:
+    """Parses the proto types to numpy types"""
+
+    nptype: np.dtype
+    feature_proto_types: List[Tuple[str, np.dtype]]
+    label_proto_type: str
+
+    @staticmethod
+    def init_types(row: serving_pb2.TrainingDataRow):
+        from featureform.serving import (
+            get_numpy_array_type,
+            proto_type_to_np_type,
+        )
+
+        types = [
+            (feature.WhichOneof("value"), proto_type_to_np_type(feature))
+            for feature in row.features
+        ]
+        numpy_array_type = get_numpy_array_type(list(map(lambda x: x[1], types)))
+        label_proto_type = row.label.WhichOneof("value")
+        return _NpProtoTypeParser(numpy_array_type, types, label_proto_type)
+
+    def parse_features(self, rows: List[serving_pb2.Value]) -> List[Any]:
+        return [
+            getattr(feature, self.feature_proto_types[i][0])
+            for i, feature in enumerate(rows)
+        ]
+
+    def parse_label(self, row: serving_pb2.Value) -> Any:
+        return getattr(row, self.label_proto_type)
+
+
+class TrainTestSplitIterator:
+    def __init__(
+        self,
+        split_stream: _SplitStream,
+        batch_size: int,
+        request_type: serving_pb2.RequestType,
+    ):
+        self.request_type = request_type
+        self._split_stream = split_stream
+        self._batch_size = batch_size
+        self._complete = False
+        self._np_type_parser = None
+
+    @staticmethod
+    def train_iter(split_stream, batch_size):
+        return TrainTestSplitIterator(
+            split_stream, batch_size, serving_pb2.RequestType.TRAINING
+        )
+
+    @staticmethod
+    def test_iter(split_stream, batch_size):
+        return TrainTestSplitIterator(
+            split_stream, batch_size, serving_pb2.RequestType.TEST
+        )
+
+    def __iter__(self):
+        return self
+
+    def __next__(self) -> Tuple[np.ndarray, np.ndarray]:
+        if self._complete:
+            raise StopIteration
+
+        batch_features_list = []
+        batch_labels_list = []
+
+        result = self._split_stream.send_request(self.request_type)
+        data = result.data
+
+        if self._np_type_parser is None and data.rows:
+            self._np_type_parser = _NpProtoTypeParser.init_types(data.rows[0])
+
+        for i, row in enumerate(data.rows):
```

Comment:
not using the index var "i". can just drop it like: 

`for row in enumerate(data.rows)`